homeassistant/components/statistics/sensor.py

```diff
@@ -390,24 +407,25 @@ def extra_state_attributes(self) -> dict[str, StateType] | None:
             key: value for key, value in self.attributes.items() if value is not None
         }
 
-    def _purge_old_states(self, max_age: timedelta) -> None:
-        """Remove states which are older than a given age."""
-        now = dt_util.utcnow()
-
-        _LOGGER.debug(
-            "%s: purging records older then %s(%s)",
-            self.entity_id,
-            dt_util.as_local(now - max_age),
-            self._samples_max_age,
-        )
+    def reset_buffer(
+        self, keep_count: int = 0, keep_age: timedelta = timedelta(-1)
+    ) -> None:
+        """Remove states, optionally keep latest according to count and/or age."""
+        self._purge_old_states(min_samples=keep_count, max_age=keep_age)
+        self._update_attributes()
+        self._update_value()
+        self.async_schedule_update_ha_state(True)
 
-        while self.ages and (now - self.ages[0]) > max_age:
-            _LOGGER.debug(
-                "%s: purging record with datetime %s(%s)",
-                self.entity_id,
-                dt_util.as_local(self.ages[0]),
-                (now - self.ages[0]),
-            )
```

Comment:
@emontnemery fyi, I would have needed to adapt those and I just didn't see the need.